src/Orleans.Runtime/GrainDirectory/GrainDirectoryHandoffManager.cs

```diff
@@ -181,15 +182,9 @@ private async void ProcessSiloStoppingEvent_Impl()
             }
             // take a copy of the current directory partition
             Dictionary<GrainId, IGrainInfo> batchUpdate = localDirectory.DirectoryPartition.GetItems();
-            try
-            {
-                await HandoffMyPartitionUponStop(batchUpdate, silosHoldingMyPartitionCopy, true);
-                localDirectory.MarkStopPreparationCompleted();
-            }
-            catch (Exception exc)
-            {
-                localDirectory.MarkStopPreparationFailed(exc);
-            }
+
+            await HandoffMyPartitionUponStop(batchUpdate, silosHoldingMyPartitionCopy, true);
```

Comment:
What is this throws?

Reply:
its caller LocalGrainDirectory catches it and `MarkStopPreparationFailed()`.

Reply:
But `localDirectory.DirectoryPartition.Clear()` will not get called as a result. Won't that cause a problem with forwarding of enqueued requests?

Reply:
yeah you are right. we need to catch it, log it  and continue

Reply:
I moved  DirectoryPartition.Clear(); to LocalGrainDirectory.Stop() after HandoffMyPartitionUponStop